webapp/resources/templates/fragments/loginProviders.html

```diff
@@ -22,7 +22,7 @@ <h3 th:utext="#{screen.welcome.label.loginwith}"/>
                     <span th:case="linkedin" class="fa fa-linkedin"></span>
                     <span th:case="facebook" class="fa fa-facebook-square"></span>
                     <span th:case="oidc" class="fa fa-openid"></span>
-                    <span th:case="saml2" class="fa fa-lock"></span>
+                    <span th:case="saml2" th:class="${entry.cssClass}" ></span>
```

Comment:
I initially had some thymeleaf code that did this, but found it a bit awkward to read; something like

`th:classappend="${#strings.replace(entry.name, ' ', '-')}"`

(but I never found out if #strings.replace took a character class).  Even though doing it in Java is a much bigger patch, it seems more logical to me.  That said, I don't really have a strong opinion on this, so if you'd rather do it that way, I can, provided the thymeleaf replace function works the way I think it should.